packages/@react-aria/link/src/useLink.ts

```diff
@@ -13,56 +13,60 @@
 import {DOMProps, PressEvent} from '@react-types/shared';
 import {HTMLAttributes, RefObject, SyntheticEvent} from 'react';
 import {LinkProps} from '@react-types/link';
-import {useId} from '@react-aria/utils';
 import {usePress} from '@react-aria/interactions';
 
 export interface AriaLinkProps extends LinkProps, DOMProps {
+  /** Whether the link is disabled. */
   isDisabled?: boolean,
-  href?: string,
-  tabIndex?: number,
-  onPress?: (e: PressEvent) => void,
-  onClick?: (e: SyntheticEvent) => void,
-  ref: RefObject<HTMLElement | null>
+  /**
+   * The HTML element used to render the link, e.g. "a", or "span".
+   * @default "a"
+   */
+  elementType?: string,
+  /** A ref to the link element. */
+  ref?: RefObject<HTMLElement | null>
 }
 
 export interface LinkAria {
+  /** Props for the link element. */
   linkProps: HTMLAttributes<HTMLDivElement>
 }
 
+/**
+ * Provides the behavior and accessibility implementation for a link component.
+ * A link allows a user to navigate to another page or resource within a web page
+ * or application.
+ */
 export function useLink(props: AriaLinkProps): LinkAria {
   let {
     id,
-    href,
     tabIndex = 0,
-    children,
+    elementType = 'a',
     onPress,
     onPressStart,
     onPressEnd,
+    // @ts-ignore
     onClick: deprecatedOnClick,
     isDisabled,
     ref
   } = props;
 
   let linkProps: HTMLAttributes<HTMLDivElement>;
-  if (typeof children === 'string') {
+  if (elementType !== 'a') {
```

Comment:
Changed this to move the children check into RSP. Seemed a bit too opinionated for react-aria. Added an `elementType` prop instead.